### PR TITLE
[RDY] Simplifying some code by re-ordering, and a fix in always setting the number of free loos.

### PR DESCRIPTION
**Pull request by Alberth289346**

### CODE REVIEW

File: CorsixTH/Lua/calls_dispatcher.lua [hunk 1]
```diff
@@ -195,26 +195,26 @@ end
...
-  end
+  local x_diff = math.abs(px - nx)
+  local y_diff = math.abs(py - ny)
+  local test_radius = 5
 
-  return staff.humanoid_class == "Nurse" and staff:isIdle() and not
-    staff:getRoom() and close_to_patient(patient,staff)
+  -- Test if the patient's room is still empty in case they are just entering
+  -- a room when they call for a staff to vaccinate them
+  return x_diff and y_diff and x_diff <= test_radius and y_diff <= test_radius
```
Comment (TheCycoONE) [inline CorsixTH/Lua/calls_dispatcher.lua]: There is no way for x_diff or y_diff  to be false here.

**Comment by TheCycoONE** [comment]
free_loos is a very misleading name for what this is. That's out of scope though and this commit looks fine.

**Comment by Alberth289346** [comment]
removed the diff_x and diff_y check, and refactors freeLoos out the code completely

**Comment by TheCycoONE** [comment]
The save game version on this now needs to be bumped. Best to rebase on latest master at the same time.

**Comment by Alberth289346** [comment]
rebased, added savegame bump

**Comment by TheCycoONE** [comment]
Bump savegame version should be merged into 95a9397

**Pull request merged**